### PR TITLE
Added implicit NSUUID JSON value transformer

### PR DESCRIPTION
**Pull request by conradev**
Let me know what you think, @robb!

### CODE REVIEW

File: Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m [hunk 1]
```diff
@@ -77,6 +78,57 @@ + (void)load {
...
 
+		MTLValueTransformer *UUIDValueTransformer = [MTLValueTransformer
+				transformerUsingForwardBlock:^id(NSString *string, BOOL *success, NSError **error) {
+					if (!string)
+						return nil;
+					
+					if (![string isKindOfClass:[NSString class]]) {
+						if (error) {
+							NSDictionary *userInfo = @{NSLocalizedDescriptionKey: NSLocalizedString(@"Could not convert string to UUID", @""),
+													   NSLocalizedFailureReasonErrorKey: [NSString stringWithFormat:NSLocalizedString(@"Expected an NSString, got: %@.", @""), string],
```
Comment (robb) [inline Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m]: Can you indent this the same as above?

File: MantleTests/MTLJSONAdapterSpec.m [hunk 2]
```diff
@@ -208,11 +208,35 @@ + (NSValueTransformer *)NSDateJSONTransformer {
...
+	
+	expect(JSONDictionary[@"URL"]).to(equal(@"http://github.com"));
+	expect(error).to(beNil());
+});
+
+
+it(@"should implicitly transform NSStrings to UUIDs", ^{
+	NSDictionary *values = @{
+	    @"UUID": @"D278C472-6DC3-4EE1-A947-861E6AF311C3",
+		@"otherUUID": @"24E1E56A-3F37-4ECE-8310-931F6ACD401A",
```
Comment (robb) [inline MantleTests/MTLJSONAdapterSpec.m]: Can you align the `@`s?

File: Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m [hunk 3]
```diff
@@ -77,6 +78,57 @@ + (void)load {
...
+						}
+						*success = NO;
+						return nil;
+					}
+					
+					return result;
+				}
+				reverseBlock:^id(NSUUID *uuid, BOOL *success, NSError **error) {
+					if (!uuid)
+						return nil;
```
Comment (robb) [inline Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m]: This `return` should be on the same line as the `if` statement.

File: Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m [hunk 4]
```diff
@@ -77,6 +78,57 @@ + (void)load {
 
 		[NSValueTransformer setValueTransformer:URLValueTransformer forName:MTLURLValueTransformerName];
 
+		MTLValueTransformer *UUIDValueTransformer = [MTLValueTransformer
+				transformerUsingForwardBlock:^id(NSString *string, BOOL *success, NSError **error) {
+					if (!string)
+						return nil;
+					
+					if (![string isKindOfClass:[NSString class]]) {
```
Comment (robb) [inline Mantle/NSValueTransformer+MTLPredefinedTransformerAdditions.m]: Can you use `NSString.class` here and below?

**Comment by robb** [comment]
Looks like great work to me.

Just some minor nits.

**Comment by conradev** [comment]
Sweet! I should've fixed the nits, let me know if you spot any more 😮

**Comment by dcaunt** [comment]
This looks good to me, any final thoughts @robb?

**Comment by robb** [comment]
:bow:

**Pull request merged**